pvlib/tests/test_pvsystem.py

```diff
@@ -1436,13 +1436,17 @@ def test__sapm_celltemp_translator():
 
 
 @fail_on_pvlib_version('0.9')
-def test_deprecated_09(cec_inverter_parameters, adr_inverter_parameters):
+def test_deprecated_09():
+    # get parameters from fixtures
+    from conftest import (cec_inverter_parameters, adr_inverter_parameters)
```

Comment:
@wholmgren can you suggest a way to use the `cec_inverter_parameters` fixture (from `conftest.py`) here?  The decorator `fail_on_pvlib_version` doesn't allow any arguments for the function to be tested, and pytest doesn't like fixtures to be imported and treated as functions.

Reply:
You can probably use a magic mock object.

Reply:
Is it possible to pass arguments through `fail_on_pvlib_version`?  [This](https://realpython.com/primer-on-python-decorators/) article suggests it could be done.

Reply:
I don’t remember if that limitation is due to my bad understanding of decorators or something about pytest interaction with a custom decorator. 

There’s no reason this function needs the working parameters, so I suspected using magic mock would be a lot faster to implement. 

I’m speculating but another option might be to turn the fail decorator into a pytest fixture.

Reply:
After further thought you'll probably need patch too:

```python
    # here we only care that warnings are emitted, 
    # so we use mock instead of real parameters
    from unittest.mock import MagicMock, patch
    cec_params = MagicMock()
    adr_params = MagicMock()
    with pytest.warns(pvlibDeprecationWarning):
        with patch('pvlib.pvsystem.snlinverter'):
            pvsystem.snlinverter(250, 40, cec_params)
```

Reply:
@wholmgren This worked for me:

https://github.com/Peque/pvlib-python/commit/44e86c20ab80fb19246058beb78a9c361ab95300

Feel free to fetch the commit from my `inverter` branch.

Reply:
Okay, *now* I had a look at https://github.com/pvlib/pvlib-python/pull/973, it seems I was late to the party... :sweat_smile: